### DISCUSSION TITLE
Allow json annotations

### DISCUSSION THREAD

**Discussion by fredt34** [geldata/gel#7077] [Ideas]
I'm using annotations for code generation - having them currenly stored as String implies some conversions. Storing json, delimited by <> or [], would be much more legible and elegant.

I found EdgeDB at v3, and it's just better and better!

**Comment by elprans** [comment]
By "some conversions" you mean `annotation foo := to_str(json_val)` and `to_json(schema::Annotation.value)`?

String is the most versatile type, because people might want to put unstructured data or plain text into an annotation, see e.g. the `std::description` annotation.  Even if arbitrarily-typed annotations were a thing (they actually used to be at some point), that wouldn't help much at introspection time, because the annotation value would have to be `str` anyway (there is no notion of scalar sum types in EdgeDB yet).

**Comment by fredt34** [comment]
Thanks for your answer - makes sense!
Yes I use annotations a lot in my POC currently, to generate code - and avoid having 2 parrallel logics in db and code.
